src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -127,26 +127,35 @@ public function getFactory()
     /**
      * Sets the service that this service is decorating.
      *
-     * @param string|null $id        The decorated service id, use null to remove decoration
-     * @param string|null $renamedId The new decorated service id
-     * @param int         $priority  The priority of decoration
+     * @param string|null $id              The decorated service id, use null to remove decoration
+     * @param string|null $renamedId       The new decorated service id
+     * @param int         $priority        The priority of decoration
+     * @param int         $invalidBehavior The behavior to adopt when decorated is invalid
      *
      * @return $this
      *
      * @throws InvalidArgumentException in case the decorated service id and the new decorated service id are equals
      */
-    public function setDecoratedService($id, $renamedId = null, $priority = 0)
+    public function setDecoratedService($id, $renamedId = null, $priority = 0/*, int $invalidBehavior = ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE*/)
     {
         if ($renamedId && $id === $renamedId) {
             throw new InvalidArgumentException(sprintf('The decorated service inner name for "%s" must be different than the service name itself.', $id));
         }
 
+        $invalidBehavior = (3 < \func_num_args()) ? (int) func_get_arg(3) : ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE;
+        if (!\in_array($invalidBehavior, [ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE, ContainerInterface::IGNORE_ON_INVALID_REFERENCE, ContainerInterface::NULL_ON_INVALID_REFERENCE])) {
```

Comment:
I don't think we need this check - we don't have in in `Reference`

Reply:
Ok, then when a call is done with an out of bounds value (`setDecoratedService('unknown_decorated', null, 0, 12)` for example), it will silently fall back to the exception behavior